homeassistant/components/garages_amsterdam/sensor.py

```diff
@@ -0,0 +1,81 @@
+"""Sensor platform for Garages Amsterdam."""
+import logging
+
+from homeassistant.const import ATTR_ATTRIBUTION
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from . import get_coordinator
+from .const import ATTRIBUTION
+
+SENSORS = {
+    "free_space_short": "mdi:car",
+    "free_space_long": "mdi:car",
+    "short_capacity": "mdi:car",
+    "long_capacity": "mdi:car",
+}
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Defer sensor setup to the shared sensor module."""
+    coordinator = await get_coordinator(hass)
+
+    async_add_entities(
+        GaragesamsterdamSensor(coordinator, config_entry.data["garage_name"], info_type)
+        for info_type in SENSORS
+    )
+
+
+class GaragesamsterdamSensor(CoordinatorEntity):
+    """Sensor representing garages amsterdam data."""
+
+    unique_id = None
+
+    def __init__(self, coordinator, garage_name, info_type):
+        """Initialize garages amsterdam sensor."""
+        super().__init__(coordinator)
+        self.unique_id = f"{garage_name}-{info_type}"
+        self.garage_name = garage_name
+        self.info_type = info_type
```

Comment:
```suggestion
        self._unique_id = f"{garage_name}-{info_type}"
        self._garage_name = garage_name
        self._info_type = info_type
```
This should allow you to remove the unique_id definition earlier.

Reply:
Obviously remember to update anywhere else that is referencing these variables.

Reply:
Thnx for the suggestions! I will change and test it 😉

Reply:
Pushed the adjustments now